From 139e924cd16515468eff92a2fec8daa8a0e0c65c Mon Sep 17 00:00:00 2001
From: saku-bruh <saku-bruh@proton.me>
Date: Thu, 31 Oct 2024 05:50:42 +0000
Subject: [PATCH] RESTRICT AUTOMERGE FRP bypass defense in App battery usage
 page

* From:
https://github.com/SomethingOS/android_packages_apps_Settings/commit/e2119ea83347b97ceaf0fe727d455434d409a507#diff-cc8d107da1df44f4a6cde3d3f5aa68c13ad5544c2f2a4e2a437c00ccc150c76bR824

Before the setup flow completion, don't allow the app info page in App battery usage to be launched.

Bug: 327748846
Test: atest SettingsRoboTests + manual test
- factory reset + launch app battery usage app info via ADB during Setup -> verify app closes
Flag : EXEMPT bugfix

(cherry picked from commit 419a6a907902a12a0f565c808fa70092004d6686)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:46724de1a273aa4110e0f645750eb48646b4546a)
Merged-In: I486820ca2afecc02729a56a3c531fb931c1907d0
Change-Id: I486820ca2afecc02729a56a3c531fb931c1907d0
---
 .../android/settings/fuelgauge/AdvancedPowerUsageDetail.java | 5 +++++
 .../settings/fuelgauge/AdvancedPowerUsageDetailTest.java     | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetail.java b/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetail.java
index d38dede..b705b56 100644
--- a/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetail.java
+++ b/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetail.java
@@ -279,6 +279,11 @@ public class AdvancedPowerUsageDetail extends DashboardFragment implements
         mLogStringBuilder = new StringBuilder("onResume mode = ").append(mOptimizationMode);
     }
 
+    @Override
+    protected boolean shouldSkipForInitialSUW() {
+        return true;
+    }
+
     @Override
     public void onPause() {
         super.onPause();
diff --git a/tests/robotests/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetailTest.java b/tests/robotests/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetailTest.java
index 5e9fb73..0a83388 100644
--- a/tests/robotests/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetailTest.java
+++ b/tests/robotests/src/com/android/settings/fuelgauge/AdvancedPowerUsageDetailTest.java
@@ -817,4 +817,9 @@ public class AdvancedPowerUsageDetailTest {
 
         verify(mBackupManager).dataChanged();
     }
+
+    @Test
+    public void shouldSkipForInitialSUW_returnTrue() {
+        assertThat(mFragment.shouldSkipForInitialSUW()).isTrue();
+    }
 }
-- 
2.43.0

